Source/Utils/Algorithms/UpperBound.H

```diff
@@ -25,12 +25,10 @@ namespace utils::algorithms
     template<typename T> AMREX_GPU_DEVICE AMREX_FORCE_INLINE
     const T* upper_bound(const T* first, const T* last, const T& val)
     {
-        const T* it;
-        size_t count, step;
-        count = last-first;
+        size_t count = last-first;
```

Comment:
We could use `amrex::upper_bound` you put into amrex.

Reply:
Actually I've checked and the algorithm is no longer used! Moreover, since it is already implemented in AMReX, I have decided to remove it: https://github.com/ECP-WarpX/WarpX/pull/4123